### PR TITLE
28149 Rewards 3.0 p2p banner

### PR DESCRIPTION
**Pull request by sujitacharya2005**
Resolves https://github.com/brave/brave-browser/issues/28149

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
**First Screen**
| Light Mode | image |  image |  image |
| :-----: | :---: | :---: | :---: |
| Dark Mode | image   | image   | image |

**Second Screen**
| Light Mode | image | image | image |
| :-----: | :---: | :---: | :---: |
| Dark Mode  | image   | image   | image   |

**All**

| Light Mode  | Dark Mode |
| ------------- | ------------- |
| image | image |
| image | image |
| image  |  image |
| image  |    image     | 
|  image  |    image  |
|  image  |  image |
|  image   | image |
|   image   |   image   |
|   image    |    image   |
|  image    |    image  |
|  image    |    image  |
|   image  |  image   |
|    image   |  image  |

**Update review comments favicon circular**
image

**Update review comments update string monthly contribution message**
image

**Review comments dark mode**
| Warning message Light Mode  | Warning message Dark Mode |
| ------------- | ------------- |
|    image   |     image    |
|   image    |    image   |
|    image      |     image    |
|  image  |   image |
|    image    |    image   |
|  image   |     image    |

**Update review comments update margin of social link buttons message**
image

### CODE REVIEW

File: browser/brave_rewards/android/brave_rewards_native_worker.h [hunk 1]
```diff
@@ -254,6 +254,7 @@ class BraveRewardsNativeWorker
   void onPublisherBanner(brave_rewards::mojom::PublisherBannerPtr wallet);
 
   void OnOneTimeTip(brave_rewards::mojom::Result result);
```
Comment (zenparsing) [inline browser/brave_rewards/android/brave_rewards_native_worker.h:256]: Can `OnOneTimeTip` be removed now?

File: browser/brave_rewards/android/brave_rewards_native_worker.h [hunk 2]
```diff
@@ -254,6 +254,7 @@ class BraveRewardsNativeWorker
...
+  void onSendContribution(bool result);
```
Comment (zenparsing) [inline browser/brave_rewards/android/brave_rewards_native_worker.h:257]: Nit: casing should match (`On...`). For this signature my suggestion would be something like:

```
void OnContributionSent(bool success);
```

**Comment by Miyayes** [comment]
Can update text to be:

>**You're already sending monthly contributions to this creator**
>You can still send a contribution now, and you can cancel your monthly contributions to this Creator by removing them from [Monthly Contributions](chrome://rewards/#monthly).

**Comment by Miyayes** [comment]
Remove dropshadow on blue button:

image

File: browser/ui/android/strings/android_brave_strings.grd [hunk 3]
```diff
@@ -142,10 +142,107 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
...
+      </message>
+      <message name="IDS__10"  desc="Text for 10">
+        10
+      </message>
+      <message name="IDS__5"  desc="Text for 5">
+        5
+      </message>
+      <message name="IDS__1"  desc="Text for 1">
+        1
+      </message>
```
Comment (mkarolin) [inline browser/ui/android/strings/android_brave_strings.grd:233]: These can be marked as `translateable=false` probably?

**Review by mkarolin — APPROVED** [review]
`strings` ++

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java [hunk 4]
```diff
@@ -0,0 +1,726 @@
...
+    }
+
+    private void setupFullHeight(BottomSheetDialog bottomSheetDialog) {
+        FrameLayout bottomSheet =
+                (FrameLayout) bottomSheetDialog.findViewById(R.id.design_bottom_sheet);
+        BottomSheetBehavior behavior = BottomSheetBehavior.from(bottomSheet);
+        behavior.setState(BottomSheetBehavior.STATE_EXPANDED);
+    }
+
+    private int getWindowHeight() {
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java:191]: I don't see where this one used, probably a leftover from previous code.

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingSuccessContributionFragment.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    private void setupFullHeight(BottomSheetDialog bottomSheetDialog) {
+        FrameLayout bottomSheet =
+                (FrameLayout) bottomSheetDialog.findViewById(R.id.design_bottom_sheet);
+        BottomSheetBehavior behavior = BottomSheetBehavior.from(bottomSheet);
+        behavior.setState(BottomSheetBehavior.STATE_EXPANDED);
+    }
+
+    private int getWindowHeight() {
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingSuccessContributionFragment.java:77]: Same here, I don't see where this one used, probably a leftover from previous code.

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java [hunk 6]
```diff
@@ -0,0 +1,726 @@
...
+        }
+        @Override
+        public void updateDrawState(@NonNull TextPaint ds) {
+            super.updateDrawState(ds);
+            ds.setUnderlineText(false);
+        }
+    };
+
+    @SuppressLint("SetTextI18n")
+    private void setLogutStateMessage() {
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java:403]: nit: setLogoutStateMessage

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingSuccessContributionFragment.java [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+        args.putDouble(AMOUNT_SELECTED, amountSelected);
+        fragment.setArguments(args);
+        return fragment;
+    }
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setStyle(STYLE_NORMAL, R.style.AppBottomSheetDialogTheme);
+    }
+    @SuppressLint("RestrictedApi")
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingSuccessContributionFragment.java:52]: I wonder the reason of this suppression, can it be the reason of crashes on older Android versions?

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java [hunk 8]
```diff
@@ -0,0 +1,726 @@
...
+                transaction.remove(fragment);
+            }
+            transaction.add(this, tag);
+            transaction.commitAllowingStateLoss();
+        } catch (IllegalStateException e) {
+            Log.e(TAG, e.getMessage());
+        }
+    }
+
+    @SuppressLint("RestrictedApi")
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java:146]: Same here, I wonder why we need it.

File: android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java [hunk 9]
```diff
@@ -1811,6 +1811,36 @@ public void onClick(View v) {
...
+            @Override
+            public void onClick(@NonNull View textView) {
+                CustomTabActivity.showInfoPage(mActivity, NEW_SIGNUP_DISABLED_URL);
+            }
+            @Override
+            public void updateDrawState(@NonNull TextPaint ds) {
+                super.updateDrawState(ds);
+                ds.setUnderlineText(false);
+            }
+        };
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java:1829]: nit: can use `NoUnderlineClickableSpan` from `chromium`

File: android/java/org/chromium/chrome/browser/rewards/tipping/MonthlyContributionToolTip.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+            }
+            return false;
+        });
+    }
+
+    private float convertDPtoPX(Context context, float dip) {
+        Resources resources = context.getResources();
+        return TypedValue.applyDimension(
+                TypedValue.COMPLEX_UNIT_DIP, dip, resources.getDisplayMetrics());
+    }
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/rewards/tipping/MonthlyContributionToolTip.java:60]: nit: Use `ViewUtils#dpToPx` chromium API. BTW, name should be `convertDptoPx`

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java [hunk 11]
```diff
@@ -0,0 +1,735 @@
...
+            currency2TextView.setText(R.string.bat);
+            usdSymbol1.setVisibility(View.INVISIBLE);
+            usdSymbol2.setVisibility(View.VISIBLE);
+            currency1ValueEditTextView.setText(String.valueOf(roundExchangeUp(usdValue)));
+            currency1ValueTextView.setText(String.valueOf(roundExchangeUp(usdValue)));
+            currency2ValueTextView.setText(String.valueOf(batValue));
+        }
+        mTipChoices[0] = mTipChoices[0];
+        mTipChoices[1] = mTipChoices[1];
+        mTipChoices[2] = mTipChoices[2];
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java:629]: Seems like not required.

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingSuccessContributionFragment.java (hunk 7 above)
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingSuccessContributionFragment.java:52]: `setupDialog` is a restricted android API method, we should use public lifecycle methods like `onCreateView` `onCreate` etc. Been updating wallet code as well in compliance with android API.

File: android/java/org/chromium/chrome/browser/rewards/tipping/TippingVerifiedCreatorToolTip.java [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+            @Override
+            public void onClick(@NonNull View textView) {
+                CustomTabActivity.showInfoPage(context, NEW_SIGNUP_DISABLED_URL);
+            }
+            @Override
+            public void updateDrawState(@NonNull TextPaint ds) {
+                super.updateDrawState(ds);
+                ds.setUnderlineText(false);
+            }
+        };
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/rewards/tipping/TippingVerifiedCreatorToolTip.java:92]: use `NoUnderlineClickableSpan` for this and similar places.

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingSuccessContributionFragment.java (hunk 7 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingSuccessContributionFragment.java]: Resolved this lint error

Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingSuccessContributionFragment.java]: used onCreateDialog

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java (hunk 8 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java]: updated

File: android/java/res/layout/brave_rewards_tippingpanel_fragment.xml [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+    <ProgressBar
+        android:id="@+id/send_tip_progress_bar"
+        android:layout_width="30dp"
+        android:layout_height="30dp"
+        android:layout_centerInParent="true"
+        app:layout_constraintTop_toTopOf="@+id/send_tip_button"
+        app:layout_constraintStart_toStartOf="@+id/send_tip_button"
+        app:layout_constraintEnd_toEndOf="@+id/send_tip_button"
```
[the remark this answers is not in the document]
Comment (sujitacharya2005) [reply android/java/res/layout/brave_rewards_tippingpanel_fragment.xml]: updated

File: browser/brave_rewards/android/brave_rewards_native_worker.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (sujitacharya2005) [reply browser/brave_rewards/android/brave_rewards_native_worker.h]: updated

**Comment by Miyayes** [comment]
- [ ] Fix dark mode to match design system
- [ ] Center social link icons in bio text area

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java [hunk 14]
```diff
@@ -0,0 +1,736 @@
...
+            return getResources().getString(R.string.gemini);
+        } else {
+            return getResources().getString(R.string.bitflyer);
+        }
+    }
+
+    private void setAlreadyMonthlyContributionSetMessage() {
+        String pubId = mBraveRewardsNativeWorker.GetPublisherId(mCurrentTabId);
+
+        boolean isPrivouslyMonthlyContributionExist =
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java:327]: Privously --> Previously

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java [hunk 15]
```diff
@@ -0,0 +1,736 @@
...
+    private boolean mIsLogoutState;
+    private double mAmountSelected;
+
+    private double rate;
+    private boolean mIsBatCurrency;
+    private BraveRewardsExternalWallet mExternalWallet;
+    private ProgressBar mTipProgressBar;
+
+    private TextView usdSymbol1;
+    private TextView usdSymbol2;
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java:108]: could you please make all fields names consistent across that new file? `usdSymbol2` --> `mUsdSymbol2`, `rate` --> `mRate` and etc. As it's very hard to understand is that var a member or a local var.

File: android/java/org/chromium/chrome/browser/rewards/tipping/TippingVerifiedCreatorToolTip.java [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+
+        ForegroundColorSpan foregroundSpan = new ForegroundColorSpan(
+                context.getResources().getColor(R.color.brave_rewards_modal_theme_color));
+        ss.setSpan(foregroundSpan, learnMoreIndex,
+                learnMoreIndex + context.getResources().getString(R.string.learn_more).length(),
+                Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        return ss;
+    }
+
+    private float convertDPtoPX(Context context, float dip) {
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/rewards/tipping/TippingVerifiedCreatorToolTip.java:107]: use an upstream function to convert dp to px

File: browser/ui/android/strings/android_brave_strings.grd [hunk 17]
```diff
@@ -142,10 +142,107 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
     <file lang="zh-TW" path="translations/android_brave_strings_zh-TW.xtb" />
   </translations>
   <release seq="1" allow_pseudo="false">
-    <messages fallback_to_english="true">
+    <messages fallback_to_english="true">    
+
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:146]: not sure why that change is done

File: browser/ui/android/strings/android_brave_strings.grd [hunk 18]
```diff
@@ -142,10 +142,107 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
...
+      <message name="IDS_CREATOR_UNABLE_RECEIVE_CONTRIBUTIONS" desc="Text for This creator is unable to receive contributions from CUSTODIAN">
+        This creator is unable to receive contributions from <ph name="CUSTODIAN">%1$s</ph>.
+      </message>
+      <message name="IDS_NOT_ENOUGH_TOKENS" desc="Text for You don't have enough tokens">
+        You don't have enough tokens
+      </message>
+      <message name="IDS_MONTHLY_RECURRING_CONTRIBUTION_TOOLTIP" desc="Text for Make this a monthly recurring contribution. You can cancel it at any time.">
+        Make this a monthly recurring contribution. You can cancel it at any time.
+      </message>
+      <message name="IDS_NOT_ENOUGH_TOKENS_DESCRIPTION" desc="Text for You don't have enough tokens">
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:162]: desc is wrong

File: browser/ui/android/strings/android_brave_strings.grd [hunk 19]
```diff
@@ -142,10 +142,107 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
...
+        You can only send up to <ph name="AMOUNT">%1$s</ph> BAT using your Brave Rewards balance.
+      </message>
+      <message name="IDS_MONTHLY_CONTRIBUTIONS" desc="Text for Monthly Contributions">
+        Monthly Contributions
+      </message>
+      <message name="IDS_ALREADY_MONTHLY_CONTRIBUTION" desc="Text for You're already sending monthly contributions">
+        You're already sending monthly contributions to this creator
+      </message>
+      <message name="IDS_ALREADY_MONTHLY_CONTRIBUTION_DESCRIPTION" desc="Text for You're already sending monthly contributions description">
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:171]: desc

File: browser/ui/android/strings/android_brave_strings.grd [hunk 20]
```diff
@@ -142,10 +142,107 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
...
+        You can still send a contribution now, and you can cancel your monthly contributions to this Creator by removing them from <ph name="MONTHLY_CONTRIBUTION">%1$s</ph>.
+      </message>
+      <message name="IDS_CONTRIBUTION_SENT" desc="Text for Contribution sent" translateable="false">
+        Contribution sent!
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:175]: I think we want to translate that

File: browser/ui/android/strings/android_brave_strings.grd [hunk 21]
```diff
@@ -142,10 +142,107 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
...
+      </message>
+      <message name="IDS_SHARE_YOUR_SUPPORT" desc="Text for Share your support" translateable="false">
+        Share your support
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:178]: I think we want to translate that

File: browser/ui/android/strings/android_brave_strings.grd [hunk 22]
```diff
@@ -142,10 +142,107 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
...
+      <message name="IDS_DOLLAR" desc="Text for dollar" translateable="false">
+        $
+      </message>
+      <message name="IDS_EMPTY_VALUE_PALCEHOLDER" desc="Text for empty value placeholder" translateable="false">
+        --
+      </message>
+      <message name="IDS_REWARDS_TIP_DEFAULT_CREATOR_DESCRIPTION" desc="Text for tipping default description">
+        Welcome! Users can support this content creator by sending them contributions. It’s a way of thanking them for making great content.
+      </message>
+      <message name="IDS_TITLE_ON_PROVIDER"  desc="Text for logged out message">
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:189]: I don't think desc is right

File: browser/ui/android/strings/android_brave_strings.grd [hunk 23]
```diff
@@ -142,10 +142,107 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
...
+        Set as monthly contribution
+      </message>
+      <message name="IDS__10"  desc="Text for 10">
+        10
+      </message>
+      <message name="IDS__5"  desc="Text for 5">
+        5
+      </message>
+      <message name="IDS__1"  desc="Text for 1">
+        1
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:232]: we don't need translate 10, 5 and 1

File: browser/ui/android/strings/android_brave_strings.grd [hunk 24]
```diff
@@ -260,92 +351,20 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
...
-      </message>
-      <message name="IDS_BRAVE_REWARDS_LOCAL_COMPLIMENT_TWEET" desc="Text of a tweet after tipping an verified creator.">
-        I just tipped <ph name="NAME">%1$s<ex>user</ex></ph> using the Brave Browser. Check it out at https://brave.com/tips.
-      </message>
-      <message name="IDS_PROBLEM_SENDING_YOUR_TIP" desc="Text for problem sending tip">
-        There was a problem sending your tip.
-      </message>
-      <message name="IDS_PROBLEM_SENDING_YOUR_TIP_DESCRIPTION" desc="Text for problem sending tip description">
-        Please close and reopen the tipping banner to try again.
+      <message name="IDS_BRAVE_REWARDS_TIP_SUCCESS_TWEET" desc="Text of a tweet after tipping an verified creator.">
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:366]: `an verified creator` --> `a verified creator`

File: browser/ui/android/strings/android_brave_strings.grd [hunk 25]
```diff
@@ -3618,9 +3616,6 @@ If you don't accept this request, VPN will not reconnect and your internet conne
     <message name="IDS_BRAVE_WALLET_SEARCHING_FOR_DOMAIN" desc="Searching for wallet domain progress message.">
       Searching for domain...
     </message>
-    <message name="IDS_REWARDS_TIPPING_SELECT_AMOUNT" desc="Select tipping amount text">
-      Please select amount to tip
-    </message>
     <message name="IDS_BRAVE_WALLET_NFTS" desc="NFTs as title">
       NFTs
     </message>
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:3621]: in general description is something that helps people translate more accurate. I don't think a description of  for example `Send amount` string should be `Text for Send amount`. It should briefly describe a context of what that string represents.

File: android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java (hunk 9 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/BraveRewardsPanel.java]: updated

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java [hunk 26]
```diff
@@ -0,0 +1,735 @@
...
+        @Override
+        public void onClick(@NonNull View textView) {
+            CustomTabActivity.showInfoPage(getActivity(), BraveActivity.BRAVE_PRIVACY_POLICY);
+        }
+        @Override
+        public void updateDrawState(@NonNull TextPaint ds) {
+            super.updateDrawState(ds);
+            ds.setUnderlineText(false);
+        }
+    };
```
[the remark this answers is not in the document]
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java]: updated

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java [hunk 27]
```diff
@@ -0,0 +1,735 @@
...
+            TabUtils.openUrlInNewTab(false, BraveActivity.BRAVE_REWARDS_SETTINGS_MONTHLY_URL);
+            getActivity().setResult(Activity.RESULT_OK);
+            getActivity().finish();
+        }
+        @Override
+        public void updateDrawState(@NonNull TextPaint ds) {
+            super.updateDrawState(ds);
+            ds.setUnderlineText(false);
+        }
+    };
```
[the remark this answers is not in the document]
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java]: updated

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java (hunk 11 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java]: updated

File: android/java/org/chromium/chrome/browser/rewards/tipping/TippingVerifiedCreatorToolTip.java (hunk 12 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/TippingVerifiedCreatorToolTip.java]: updated

File: android/java/org/chromium/chrome/browser/rewards/tipping/MonthlyContributionToolTip.java (hunk 10 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/MonthlyContributionToolTip.java]: updated

**Comment by sujitacharya2005** [comment]
> * [x]  Fix dark mode to match design system
> * [x]  Center social link icons in bio text area

@chrismore This is fixed

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java (hunk 14 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java]: updated

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java (hunk 15 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java]: updated

File: android/java/org/chromium/chrome/browser/rewards/tipping/TippingVerifiedCreatorToolTip.java (hunk 16 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/TippingVerifiedCreatorToolTip.java]: updated

File: browser/ui/android/strings/android_brave_strings.grd (hunk 22 above)
Comment (sujitacharya2005) [reply browser/ui/android/strings/android_brave_strings.grd]: updated

File: browser/ui/android/strings/android_brave_strings.grd [hunk 28]
```diff
@@ -142,10 +142,107 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
...
+      <message name="IDS_REWARDS_TIP_DEFAULT_CREATOR_DESCRIPTION" desc="Text for tipping default description">
+        Welcome! Users can support this content creator by sending them contributions. It’s a way of thanking them for making great content.
+      </message>
+      <message name="IDS_TITLE_ON_PROVIDER"  desc="Text for logged out message">
+        <ph name="TITLE">%1$s</ph> on <ph name="PROVIDER">%2$s</ph>
+      </message>
+      <message name="IDS_LOGGED_OUT_OF_CUSTODIAN"  desc="Text for logged out message">
+        Logged out of <ph name="CUSTODIAN">%1$s</ph>
+      </message>
+      <message name="IDS_LOGIN_TO_CUSTODIAN"  desc="Text for logged out message">
```
[the remark this answers is not in the document]
Comment (sujitacharya2005) [reply browser/ui/android/strings/android_brave_strings.grd]: updated

File: browser/ui/android/strings/android_brave_strings.grd [hunk 29]
```diff
@@ -142,10 +142,107 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
...
+        Log in to <ph name="CUSTODIAN">%1$s</ph>
+      </message>
+      <message name="IDS_THANKS_FOR_SUPPORTING_YOUR_FAVORITE_CREATORS" desc="Text for Thanks for supporting your favorite creators">
+        Thanks for supporting your favorite creators
+      </message>
+      <message name="IDS_LOGGED_OUT_OF_CUSTODIAN_DESCRIPTION"  desc="Text for logged out message description">
+        It looks like you're logged out of <ph name="CUSTODIAN">%1$s</ph> for Brave Rewards. Log in again to continue.
+      </message>
+      <message name="IDS_INFO_CREATOR_NOT_VERIFIED"  desc="Text for Send contribution">
```
[the remark this answers is not in the document]
Comment (sujitacharya2005) [reply browser/ui/android/strings/android_brave_strings.grd]: updated

File: browser/ui/android/strings/android_brave_strings.grd (hunk 23 above)
Comment (sujitacharya2005) [reply browser/ui/android/strings/android_brave_strings.grd:232]: updated

File: browser/ui/android/strings/android_brave_strings.grd (hunk 24 above)
Comment (sujitacharya2005) [reply browser/ui/android/strings/android_brave_strings.grd]: updated

File: browser/ui/android/strings/android_brave_strings.grd (hunk 17 above)
Comment (sujitacharya2005) [reply browser/ui/android/strings/android_brave_strings.grd:146]: updated

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java (hunk 15 above)
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java]: Please update all members in that file
```
    private TextView currency1TextView;
    private TextView currency2TextView;
    private EditText currency1ValueEditTextView;
    private TextView currency1ValueTextView;
    private TextView currency2ValueTextView;
```

Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java]: updated

**Review by SergeyZhukovsky — APPROVED** [review]
++

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java (hunk 6 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java]: updated

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingSuccessContributionFragment.java (hunk 5 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingSuccessContributionFragment.java]: updated

**Review by Pavneet-Sing — APPROVED** [review]

File: browser/brave_rewards/android/brave_rewards_native_worker.h (hunk 1 above)
Comment (sujitacharya2005) [reply browser/brave_rewards/android/brave_rewards_native_worker.h]: removed

**Review by deeppandya — APPROVED** [review]
LGTM

**Pull request merged**